### PR TITLE
Fix schedule and then publish flow

### PR DESCRIPTION
**Pull request by oandregal**
Fixes https://github.com/WordPress/gutenberg/issues/10930

### CODE REVIEW

File: packages/editor/src/components/post-publish-panel/index.js
Comment (nosolosw) [inline packages/editor/src/components/post-publish-panel/index.js]: :tada:

File: packages/editor/src/components/post-publish-panel/index.js
Comment (noisysocks) [inline packages/editor/src/components/post-publish-panel/index.js]: It doesn't look like `postType` is used anywhere.

File: packages/editor/src/components/post-publish-panel/test/index.js
Comment (noisysocks) [inline packages/editor/src/components/post-publish-panel/test/index.js]: Typo: `s/is post status/if post status/`

File: packages/editor/src/components/post-publish-panel/test/__snapshots__/index.js.snap
Comment (noisysocks) [inline packages/editor/src/components/post-publish-panel/test/__snapshots__/index.js.snap]: We're only asserting that a function named `_default` is rendered here, which isn't quite right.

This is because of how we define the `Spinner` component:

```js
export default () => <span className="components-spinner" />;
```

It should probably be:

```js
export default function Spinner() {
	return <span className="components-spinner" />;
}
```

**Review by noisysocks — APPROVED** [review]
This is looking great! When I test this locally I'm unable to reproduce the bug by following the steps in #10930.

Thanks for adding unit tests and for simplifying `PostPublishPanel`. It's a lot easier to understand how the component works now that there's no state 😍🌈

I left some minor comments which should be fixed up before merging. Giving you the 👍 though so that you're not blocked on me re-reviewing.

File: packages/editor/src/components/post-publish-panel/test/index.js
Comment (nosolosw) [reply packages/editor/src/components/post-publish-panel/test/index.js]: ae84bd4

File: packages/editor/src/components/post-publish-panel/test/__snapshots__/index.js.snap
Comment (nosolosw) [reply packages/editor/src/components/post-publish-panel/test/__snapshots__/index.js.snap]: 8a6c0bc

File: packages/editor/src/components/post-publish-panel/index.js
Comment (nosolosw) [reply packages/editor/src/components/post-publish-panel/index.js]: 2bac72b

File: packages/editor/src/components/post-publish-panel/index.js
Comment (nosolosw) [inline packages/editor/src/components/post-publish-panel/index.js]: Removed, was a leftover.

File: packages/editor/src/components/post-publish-panel/index.js
Comment (nosolosw) [inline packages/editor/src/components/post-publish-panel/index.js]: This was the bug: we were using only `isScheduled` to determine whether a post is scheduled, but we also need to take into account whether the date scheduled is > now (which is what `isBeingScheduled` does).

File: packages/editor/src/components/post-publish-panel/index.js
Comment (nosolosw) [inline packages/editor/src/components/post-publish-panel/index.js]: This logic wasn't accounting for the fact that an user can:

- schedule a post with a date in the future
- change its mind and modify the date to be in the past: after this, the user should be able to publish directly but wasn't.

**Pull request merged**
